packages/api-core/src/graphql/schema.graphql

```diff
@@ -34,6 +34,100 @@ enum MassUnit {
   oz
 }
 
+"Relational Operator Types used in filtering inside a single condition"
+enum RelationalOperatorTypes{
+  "Begins With used with String types"
+  beginsWith
+
+  "Ends With used with String types"
+  endsWith
+
+  "Equal"
+  eq
+
+  "Greater Than"
+  gt
+
+  "Greater Than or Equal"
+  gte
+
+  "In"
+  in
+
+  "Less Than"
+  lt
+
+  "Less Than or Equal"
+  lte
+
+  "Not Equal"
+  ne
+
+  "Not In"
+  nin
+
+  "Regex"
+  regex
+}
+
+"Single Condition for filterSearch, use exactly one of the optional input value type"
+input SingleConditionInput {
+  "Value to filter if it is Boolean input"
+  boolValue: Boolean
+
+  "Flag to set if the regex is case insensitive"
+  caseSensitive: Boolean
+
+  "Value to filter if it is Date input"
+  dateValue: DateTime
+
+  "Value to filter if it is Float Array input" 
+  floatArrayValue: [Float]
+
+  "Value to filter if it is Float input"
+  floatValue: Float
+
+  "Value to filter if it is Int Array input" 
+  intArrayValue: [Int]
+
+  "Value to filter if it is Int input"
+  intValue: Int
+
+  "Field name"
+	key : String!
+
+  "Logical NOT operator to negate the condition"
+  logicalNOT: Boolean
+
+  "Relational Operator to join the key and value"
+  relationalOperator: RelationalOperatorTypes!
+
+  "Value to filter if it is String Array input"
+  stringArrayValue: [String]
+
+  "Value to filter if it is String input"
+	stringValue: String
+}
+
+"Filter search with One level of conditions (use either 'any' or 'all' not both)"
+input ConditionsArray {
+  "Array of single-conditions"
+  all: [SingleConditionInput]
+
+  "Array of single-conditions"
+  any: [SingleConditionInput]
+}
+
+"Filter search with nested conditions of input (use either 'any' or 'all' not both)"
+input FilterConditionsInput {
+  "Array holding Nested conditions (use either 'any' or 'all' not both)"
+  all: [ConditionsArray]
+
+  "Array holding Nested conditions (use either 'any' or 'all' not both)"
+  any: [ConditionsArray]
```

Comment:
This looks like it forces you to have 2 levels, but often you would want to have only 1 level. I believe unions are still not supported for `input` types, but you could do something like this:

```graphql
input FilterConditionsInput {
  "Array of single-conditions"
  all: [SingleConditionInput]

  "Array of single-conditions"
  any: [SingleConditionInput]

  "Array holding Nested conditions (use either 'any' or 'all' not both)"
  allNested: [ConditionsArray]

  "Array holding Nested conditions (use either 'any' or 'all' not both)"
  anyNested: [ConditionsArray]
}
```

I don't love it but it's all I can think of right now.

Reply:
We actually considered that approach and decided that the forced nesting was the less confusing of the two options.